src/reconcile/rec_write.c

```diff
@@ -1069,33 +1069,33 @@ __rec_split_row_promote(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_ITEM *key,
      * key.
      */
     max = r->last;
-    if (r->cache_write_restore)
```

Comment:
@michaelcahill @keitharnoldsmith I don't understand the purpose of this. But I think there is another problem here, we should do key promotion all the time as we save updates both in eviction and checkpoint.

Reply:
I'm guessing you meant to include @keithbostic on this?

Reply:
My understanding is that this change will promote keys during checkpoint reconciliation, which  makes sense to me.

Reply:
Checkpoint doesn't modify content in cache so we shouldn't need to worry about it needing to do key promotion here.

Reply:
Also I think we can leave this change out as part of 5826? Am I right in thinking that? I'm not too comfortable with making changes to code we don't fully understand especially if they don't relate to 5826.

Reply:
Yes, we can leave it out for now. @luke-pearson Are you sure this is not required for checkpoint? I'll create a separate ticket to investigate this.

Reply:
Lets discuss the checkpoint key promotion separately and remove this change form this PR.

Reply:
Removed as discussed.